I2CRotaryEncoder/I2CRotaryEncoder.ino

```diff
@@ -6,53 +9,70 @@
 #define I2C_ROTARY_SIGNATURE    0x67
 #define I2C_ROTARY_VALUE        0x65
 
-const byte encoderPinA = 2;
-const byte encoderPinB = 3;
-
-volatile int count = 0;
+#define DEBUG 1         //level 1 für Seriellen Monitor, 0 für aus
 
 uint8_t encoder_value = 0;
-uint8_t encoder_value_prev = 0;
-
-#define readA bitRead(PIND,2) // Faster to read this way than digitalRead();
-#define readB bitRead(PIND,3) // Faster to read this way than digitalRead();
 
 void setup() {
-  pinMode(encoderPinA, INPUT_PULLUP);
-  pinMode(encoderPinB, INPUT_PULLUP);
-  attachInterrupt(digitalPinToInterrupt(encoderPinA), isrA, FALLING);
-  attachInterrupt(digitalPinToInterrupt(encoderPinB), isrB, FALLING);
+  pinMode(CLK, INPUT);
+  pinMode(CLK, INPUT_PULLUP);
+  pinMode(DATA, INPUT);
+  pinMode(DATA, INPUT_PULLUP);
   Wire.begin(I2C_ADDRESS);
   Wire.onRequest(requestEvent);
   Wire.onReceive(receiveEvent);
-  
-} 
-
-void loop()  {
-  noInterrupts();
-  encoder_value = count;
-  interrupts();
-  encoder_value_prev = encoder_value;
+      if (DEBUG >= 1){
+        Serial.begin (115200);
+        Serial.println("I2C Rotaryencoder for Tasmota");
+      }
 }
 
-void isrA() {
-  if(readB != readA) {
-    count ++;
-  } else {
-    count --;
-  }
-  if (count < 0) count=0;
-  if (count > 100) count=100;
```

Comment:
no limitation (Value could be 0 - 255) turn should be evaluated by f.e. OpenHab (smartHome server)